src/main/scala/org/broadinstitute/dsde/workbench/sam/azure/AzureRoutes.scala

```diff
@@ -57,6 +57,36 @@ trait AzureRoutes extends SecurityDirectives with LazyLogging with SamRequestCon
                 }
               }
             } ~
+            pathPrefix("actionManagedIdentity") {
+              path(Segment / Segment / Segment) { (resourceTypeName, resourceId, action) =>
+                val resource = FullyQualifiedResourceId(ResourceTypeName(resourceTypeName), ResourceId(resourceId))
+                val resourceAction = ResourceAction(action)
+
+                withNonAdminResourceType(resource.resourceTypeName) { resourceType =>
+                  if (!resourceType.actionPatterns.map(ap => ResourceAction(ap.value)).contains(resourceAction)) {
+                    throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"action $action not found"))
+                  }
+                  pathEndOrSingleSlash {
+                    postWithTelemetry(
+                      samRequestContext,
+                      "resourceType" -> resource.resourceTypeName,
+                      "resource" -> resource.resourceId,
+                      "action" -> resourceAction
+                    ) {
+                      entity(as[ManagedResourceGroupCoordinates]) { mrgCoordinates =>
```

Comment:
Should the input be a billing profile id instead of full coordinates? Sam should already have the coordinates in the DB. Then we can be sure the MRG is kosher.